python/update_package_properties.py

```diff
@@ -0,0 +1,297 @@
+import argparse
+import re
+from datetime import date, datetime
+
+import pathlib2
+import string_utils
+from github import Github
+from jinja2 import Environment, FileSystemLoader
+from parameters_validation import (no_whitespaces, non_blank, non_empty, non_negative, validate_parameters,
+                                   parameter_validation)
+from dataclasses import dataclass
+
+BASE_PATH = pathlib2.Path(__file__).parent.absolute()
+
+
+@parameter_validation
+def is_version(version: str):
+    if version is None or not version:
+        raise ValueError("version should be non-empty and should not be None")
+    if not re.match(r"\d+\.\d+\.\d+$", version):
+        raise ValueError(
+            "version should include 3 levels of versions consists of numbers  separated with dots. e.g: 10.0.1")
+
+
+@parameter_validation
+def is_tag(tag: str):
+    if tag is None or not tag:
+        raise ValueError("tag  should be non-empty and should not be None")
+    if not re.match(r"v\d+\.\d+\.\d+$", tag):
+        raise ValueError(
+            "tag  should start with v and should include 3 levels of versions consists of numbers " +
+            "separated with dots. e.g: v10.0.1")
+
+
+@parameter_validation
+def is_email(email: str):
+    if not string_utils.is_email(email):
+        raise ValueError("Parameter is not in email format")
+
+
+@parameter_validation
+def is_project_changelog_header(header: str):
+    if header is None or not header:
+        raise ValueError("header should be non-empty and should not be None")
+    if not re.match(r"^### \w+\sv\d+\.\d+\.\d+\s\(\w+\s\d+,\s\d+\)\s###$", header):
+        raise ValueError(
+            f"changelog header is in invalid format. Actual:{header} Expected: ### citus v8.3.3 (March 23, 2021) ### ")
+
+
+@dataclass
+class ChangelogParams:
+    latest_changelog: str = ""
+    project_version: str = ""
+    project_name: str = ""
+    fancy: bool = False
+    fancy_version_number: int = 0
+    microsoft_email: str = ""
+    name_surname: str = ""
+    changelog_date: datetime = datetime.now()
+
+
+def get_spec_file_name(project_name: str) -> str:
+    return f"{project_name}.spec"
+
+
+def get_version_number(version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}{fancy_suffix}"
+
+
+def get_rpm_version(project_name: str, version: str) -> str:
+    return f"{version}.{project_name}"
+
+
+def get_version_number_with_project_name(project_name: str, version: str, fancy: bool,
+                                         fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}.{project_name}{fancy_suffix}"
```

Comment:
These kind of functions would be better as methods of `ChangelogParams`, also prefixing functions with `get_` is not very common in Python.

```suggestion
	  def spec_file_name(self) -> str:
	      return f"{self.project_name}.spec"
	  
	  
	  def version_number(self) -> str:
	      fancy_suffix = f"-{self.fancy_release_number}" if self.fancy else ""
	      return f"{self.project_version}{fancy_suffix}"
	  
	  
	  def rpm_version() -> str:
	      return f"{self.project_version}.{self.project_name}"
	  
	  
	  def version_number_with_project_name(self) -> str:
	      fancy_suffix = f"-{self.fancy_release_number}" if self.fancy else ""
	      return f"{self.project_version}.{self.project_name}{fancy_suffix}"
```

Reply:
BTW. there could be more of these functions that could be better as methods of `ChangelogParams`. But these ones stood out to me a lot.

Reply:
Fixed

Reply:
cb341d202648b2be0da0cac24cb818c05ccfe8d5